core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -90,4 +101,65 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
   protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
+
+  @Override
+  public boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    boolean valid = test( connectionDetails );
+    if ( !valid ) {
+      return false;
+    }
+
+    if ( !connectionDetails.isSupportsRootPath()  || connectionTestOptions.isIgnoreRootPath() ) {
+      return true;
+    }
+
+    String resolvedRootPath = getResolvedRootPath( connectionDetails );
+    if ( StringUtils.isEmpty( resolvedRootPath ) ) {
+      return !connectionDetails.isRootPathRequired();
+    }
+
+    String internalUrl = buildUrl( connectionDetails, resolvedRootPath );
+    FileObject fileObject = KettleVFS.getFileObject( internalUrl, new Variables(), getOpts( connectionDetails ) );
+
+    try {
+      return fileObject.exists();
```

Comment:
It seems we are testing twice, It's related to what rootPath is? https://github.com/pentaho/pentaho-kettle/pull/9305/files#r1589437196

I'm not saying it's right or wrong.

Reply:
Validating the VFS credentials before proceeding to verify root path in test(ConnectionDetails, VFSConnectionTestOptions) . 

Will it not be fine if we retest the connection details again before validating root path? Please suggest.

Reply:
It's fine, it's just from a User's perspective. They won't know either way. Only thing I could think of maybe adding a generic log statement for both tests that don't disclose the root path value.

Either way it's not blocking comment to merge the PR

Reply:
The original `#test(ConnectionDetails)` implementations, which exist in each connection provider class, test access to the real root path, `/`.
The new overload, `#test(ConnectionDetails, VFSConnectionTestOptions)` allows taking into account the root path, when `VFSConnectionTestOptions#isIgnoreRootPath()` is `true`.

We can interpret this as the default testing, when the options argument is not passed, not including the root path.

I would prefer the default to include the root path. Maybe this is the (main) point you're making.
However, we'd have to break compatibility with / modify the current `#test(ConnectionDetails)` implementations. Only a real issue for the any third-party implementations out there whose code we wouldn't be able to modify.
We have been ignoring backward compatibility breakages in other instances, though, so maybe we should revisit this decision here.

Also, I did not understand what you meant with adding a log statement. Not including the root path in validation is an input of the method. I don't think we should be logging a warning when this is `true` out of fear of misuse.

@NJtwentyone All in all, what do you think?